static/src/javascripts/projects/common/modules/ui/subscription-banners/guardian-weekly-banner.spec.js

```diff
@@ -0,0 +1,280 @@
+// @flow
+
+import { bannerTemplate as gwBannerTemplate } from './guardian-weekly-banner-template';
+import { selectorName, createBannerShow } from './subscription-banner';
+
+jest.mock('ophan/ng', () => null);
+jest.mock('common/modules/commercial/user-features.js', () => jest.fn());
+jest.mock('common/modules/commercial/contributions-utilities.js', () => ({
+    getReaderRevenueRegion: jest.fn().mockReturnValue('united-kingdom'),
+}));
+jest.mock('common/modules/identity/api.js', () => ({
+    isUserLoggedIn: jest.fn().mockReturnValue(true),
+}));
+jest.mock('lib/report-error', () => jest.fn());
+
+const removeAllSpaces = (str: string) => str.replace(/\s/g, '');
+
+describe('Guardian Weekly Banner', () => {
```

Comment:
what's the value of testing this and the digital banner separately? and if there is value, then could we combine the 2 sets of tests and just run them twice, once for `gwBannerTemplate` and once for the other template

Reply:
Because the banners can change independently, I though it would be wise to test both of them to protect against changes that are made to them at different times